projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+)
+
+func main() {
+	res, err := http.Get("http://localhost:8080")
+	if err != nil {
+
+		if res != nil {
```

Comment:
_Most_ (but not all!) functions in Go that return a value and an error only ever set one - i.e. `err` and `res` will never _both_ be non-nil. You generally want to handle the "error is not nil" (i.e. there was an error) case _separately_ from the "res is not nil" (i.e. there was not error, there is a value) case, rather than putting one inside the other.